### PR TITLE
NotificationDetails + Swift Mk3: Resources

### PR DESCRIPTION
**Pull request by jleandroperez**
#### Details:

Migrating **NotificationDetailsViewController**, step by step!.
Reference #5613

Needs Review: @kwonye 

Sir, may i bug you with yet another review?

Thanks **A LOT** in advance!!!

--

#### Scenario A: Followers
1. Log into a dotcom account, and open a *New Follower* Notification.
2. Press over the Notification's Header
 3. Verify that the `Site's Stream` shows up (Native reader, with `Follow` / `Unfollow` capabilities on top).
4. Press over the Notification's Footer (See all of your followers)
 5. Please, note that this will only be only visible if you've got +20 followers.
 6. Verify that the `Native Followers List` shows up onscreen.

Please, DM me, i can send you over a testing account!

#### Scenario B: Comments
1. Open a *Comment-Y* Notification. Please **pick a Notification you've already replied to**.
2. Tap over the header.
 3. Verify that the `Native Comments Reader` shows up onscreen.
4. Press over the `You have replied this comment` legend (at the bottom, before the actions row)
 5. Verify that the `Native Comments Reader` shows up onscreen.

#### Scenario C: Likes
1. Log into a dotcom account, and open a *Post Like* Notification
2. Tap over the header
 3. Verify that the `Native Reader` shows up onscreen.
4. Tap over any of the user's cells
 5. Verify that the `Stream Reader` shows up onscreen.

#### Scenario D: Attachments
1. Open a Comment Notification (with an image attach + links!)
2. Verify that pressing the Image opens the `Native Image Viewer`, fullscreen, with the flickable behavior
3. Verify that pressing any of the links opens a WebView
4. Verify that pressing the Author's blog title pushes the Native Reader

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift [hunk 1]
```diff
@@ -478,6 +478,202 @@ extension NotificationDetailsViewController
 
 
 
+/// Resources
+///
+extension NotificationDetailsViewController
+{
+    func displayURL(url: NSURL?) {
+        guard let url = url else {
```
Comment (kwonye) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift]: Great job cleaning up `displayURL` when converting from `openURL` 👍

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift [hunk 2]
```diff
@@ -478,6 +478,202 @@ extension NotificationDetailsViewController
...
+
+    private func displayResourceWithRange(range: NotificationRange?) throws {
+        guard let range = range else {
+            throw DisplayError.MissingParameter
+        }
+
+        switch range.type {
+        case .Site:
+            try displayStreamWithSiteID(range.siteID)
+
```
Comment (kwonye) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift]: For consistency's sake, can we remove the spacing between the case statements please? None of the other switch statements in the app have spaces between them.

Comment (kwonye) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift]: most* (at least none in Swift that I saw)

**Comment by kwonye** [comment]
Unfortunately inline images are not working.

**Comment by kwonye** [comment]
:shipit:

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift (hunk 1 above)
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift]: Thank you!! LONG deserved cleanup, hopefully, much more readable!!!

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift (hunk 2 above)
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift]: Done, thank you!!

**Comment by jleandroperez** [comment]
**Pure awesomeness** @kwonye !!!

Thank you!!!

**Pull request merged**